### PR TITLE
Codegen Backend Service API interfaces

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1203425544997318/f

### Description
Codegen the Retrofit service APIs. See [asana task](https://app.asana.com/0/488551667048375/1203425544997318/f) for more info.
Also created a lint rule to detect calls to `retrofit2.Retrofit.create()` method and force the use of the codegen annotation.

### Steps to test this PR

- [x] PR builds
- [x] smoke tests for the app
- [x] test StatisticsService 
- [x] test AppTrackerListService 
- [x] test FeedbackService 
- [x] test HttpsUpgradeService 
- [x] test ResourceSurrogateListService 
- [x] test SurveyService
- [x] test TrackerListService
- [x] test DownloadFileService
- [x] test PrivacyConfigService
- [x] test RemoteMessagingService
- [x] test PixelService
- [x] test AutoCompleteService
- [x] test EmailService

_Note: filtering logcat by `Downloader` should allow to see the logs for every downloader that users the above services_

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2554** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2554" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2554?utm_source=stack-comment).

File: app/src/main/java/com/duckduckgo/app/di/NetworkModule.kt [hunk 1]
```diff
@@ -150,30 +146,16 @@ class NetworkModule {
...
 
-    @Provides
-    fun trackerListService(@Named("api") retrofit: Retrofit): TrackerListService =
-        retrofit.create(TrackerListService::class.java)
-
-    @Provides
-    fun httpsUpgradeService(@Named("api") retrofit: Retrofit): HttpsUpgradeService =
-        retrofit.create(HttpsUpgradeService::class.java)
-
+    @SuppressLint("NoRetrofitCreateMethodCallDetector")
```
Comment (CDRussell) [inline app/src/main/java/com/duckduckgo/app/di/NetworkModule.kt:149]: What makes next for the non-caching services... should we also have a `ContributesNonCachingServiceApi` annotation?

Comment (marcosholgado) [reply app/src/main/java/com/duckduckgo/app/di/NetworkModule.kt:149]: Could we pass the named parameter? i.e `@ContributesServiceApi(AppScope::class, named="api")` In my head that's better than copy&pasting the codegen for each different type.

Comment (CDRussell) [reply app/src/main/java/com/duckduckgo/app/di/NetworkModule.kt:149]: For `AutoCompleteService`, we want to ensure those results are not cached as they potentially contain sensitive data (like your search queries).

More generally, the idea was to separate the requests into two types:
- non-sensitive requests (like downloading config files) where it's safe to keep this even after clearing all data
- sensitive requests we shouldn't preserve

Originally added here for context. https://github.com/duckduckgo/Android/pull/515

Comment (aitorvs) [reply app/src/main/java/com/duckduckgo/app/di/NetworkModule.kt:149]: This make sense. I will also migrate the non caching then. Neither are irreversible changes plus having both is good for  consistency I guess.

**Comment by aitorvs** [comment]
@CDRussell, @marcosholgado expanded the codege to consider nonCaching APIs in the latest commit https://github.com/duckduckgo/Android/pull/2554/commits/ef9894da7b79a495729e8165e362cacda5e4b633

**Review by marcosholgado — APPROVED** [review]
LGTM

**Pull request merged**